### DISCUSSION TITLE
How to get HTML report out of starboard-operator?

### DISCUSSION THREAD

**Discussion by strus38** [aquasecurity/starboard#615] [Questions & Help] [answered]
Hi
I have run successfully startboard-operator in the cluster, and I get all the results wanted. However, I need to expose them to people outside, and I am looking at a way to procude a HTML eexport of those.
any suggestion is welcome.
thanks.

**Comment by danielpacak** [comment, marked as answer]
Starboard Operator is responsible for discovering K8s resources, scheduling scanners, and finally saving data as custom resources. It **does not** provide any UI or HTTP service to generate HTML reports.

However, in some cases you may use Starboard CLI as a compliment of the operator's functionality and use it to export custom resources to HTML. Alternatively, you can implement your own export tool that reads Starboard custom resources and transform them into HTML pages.

Below you can find examples of Starboard CLI commands to generate HTML reports for certain Kubernetes core objects:

```
# Generate an HTML report for a deployment with the specified name and save it to a file.
kubectl starboard get report deployment/nginx > nginx.deploy.html

# Generate an HTML report for a namespace with the specified name and save it to a file.
kubectl starboard get report namespace/kube-system > kube-system.ns.html

# Generate an HTML report for a node with the specified name and save it to a file.
kubectl starboard get report node/kind-control-plane > kind-control-plane.node.html
```

Find more details by displaying help for the `get report` command:

```
starboard get report --help
```

**Comment by rafaribe** [comment]
Thanks for the clarification. Is there any plan to integrate a dashboard of some sort so we can check the report results in a more human-friendly way? I'm thinking something like the Harbor Vulnerability scanner page that also uses `trivy` as the scanner.

**Reply by danielpacak** [reply]
I don't think we'll provide an open source, standalone dashboard or web app to visualise security reports generated by Starboard and its integrated scanners. The reason for that is that we're not UI / UX experts and we'd rather focus on actionable security reports. However, we do integrate with popular Kubernetes IDEs / dashboards such as https://github.com/aquasecurity/starboard-octant-plugin and https://github.com/aquasecurity/starboard-lens-extension so you can access reports in a more human-friendly way.
